scalyr_agent/scalyr_logging.py

```diff
@@ -472,6 +486,14 @@ def _log(
         else:
             __thread_local__.last_error_for_monitor = None
 
+        stdout_handler = self.__stdout_handler
+        stderr_handler = self.__stderr_handler
+
+        if force_stdout and stdout_handler:
```

Comment:
Rather than adding and removing this handler for each `emit`, we should just  re-use the `__thread_local__` pattern here.  We should have `__thread_local__.last_force_stdout` and `__thread_local__.last_force_stderr` that get set here.  Then, use that in `makeRecord` to set attributes `force_stdout` and `force_stderr` on the returned record.  Finally, add a filter to the stdout and stderr handlers such that they are only used if the record they are asked to handle has `force_stdout` or `force_stderr` set to True.

Reply:
That would cover any possible race conditions too (I don't think there are any but better safe than sorry), sounds good to me.